[MINOR] docs: --refine the how to install guide (#5726)
apache/gravitino: docs/how-to-install.md

## Install Apache Gravitino from scratch

### Prerequisites

> If you have not installed Java, install it and configure the `JAVA_HOME` environment variable. After installation, you can run the `${JAVA_HOME}/bin/java -version` command to check the Java version.

Reviewer: The convention is to treat docs as code, and to use git for change tracking.
Since git works on a line-by-line comparison, we'd better wrap the docs manually
so that changes can be easily tracked and reviewed.

```suggestion
If you have not installed Java, install it and configure the `JAVA_HOME` environment variable.
After the installation, you can run the `${JAVA_HOME}/bin/java -version` command to check the Java version.
```

Author: Got it. check through the whole doc and fix it. PTAL

        |── conf/                                   # All configurations for Gravitino.
        |   ├── gravitino.conf                      # Gravitino server and Gravitino Iceberg REST server configuration.
        |   ├── gravitino-iceberg-rest-server.conf  # Gravitino server configuration.
        |   ├── gravitino-env.sh                    # Environment variables, etc., JAVA_HOME, GRAVITINO_HOME, and more.
        |   └── log4j2.properties                   # log4j configuration for the Gravitino server and Gravitino Iceberg REST server.
        |── libs/                                   # Gravitino server dependencies libraries.
        |── logs/                                   # Gravitino server and Gravitino Iceberg REST server logs. Automatically created after the server starts.
        |── data/                                   # Default directory for the Gravitino server to store data.
        |── iceberg-rest-server/                    # Gravitino Iceberg REST server package and dependencies libraries.
        └── scripts/                                # Extra scripts for Gravitino.
    ```

### Initialize the RDBMS (Optional)

> If you want to use the relational backend storage, you need to initialize the RDBMS. For details on initializing the RDBMS, see [How to use relational backend storage](./how-to-use-relational-backend-storage.md).

Reviewer: ```suggestion
If you want to use the `relational` backend storage, you need to initialize the RDBMS. 
For details on initializing the RDBMS, see
[How to use relational backend storage](./how-to-use-relational-backend-storage.md).
```

Author: Ditto

### Configure the Gravitino server

> The `conf/gravitino.conf` file provides basic configurations of the Gravitino server. To configure the Gravitino server, you can update this file. For a full list of the Gravitino server configurations, see [Gravitino Server Configurations](./gravitino-server-config.md).

Reviewer: ```suggestion
The `conf/gravitino.conf` file provides basic configurations for the Gravitino server. 
To configure the Gravitino server, you can edit this file.
For a full list of the Gravitino server configurations, see
[Gravitino Server Configurations](./gravitino-server-config.md).
```

Author: Ditto

### Configure the Gravitino server log

Gravitino uses [Log4j2](https://logging.apache.org/log4j/2.x/) as the logging system. The `conf/log4j2.properties` file provides the Gravitino server log configurations. To configure the Gravitino server log, you can update this file.

### Configure the Gravitino server environment variables

Gravitino exposes several environment variables through the `conf/gravitino-env.sh` file. To configure these environment variables, you can update this file.

### Configure the Gravitino catalogs

> 

Reviewer: Better add a brief introduction of 'catalog provider'.

Author: good suggestion. Will consider to add it when i get familiar with the product and the docs

Gravitino supports multiple catalogs. You can configure the catalog-level configurations by updating the related configuration file in the `catalogs/<catalog-provider>/conf` directory. For example, the `catalogs/hive/conf/hive.conf` file provides configurations for the Hive catalog. The configurations you set in the catalog configuration file will apply to all the catalogs of the same type that you created. For detailed configurations about each catalog, see related catalog documentation.

:::note
Gravitino takes the catalog configurations in the following order of precedence:

1. Catalog `properties` specified in the catalog creation API or REST API.
2. Catalog configurations specified in the catalog configuration file.

The catalog `properties` overrides the catalog configurations specified in the configuration file.
:::

> Gravitino supports passing in catalog-specific configurations by adding the `gravitino.bypass.` prefix to the specific catalog configurations. For example, you can pass in the HMS-specific configuration `hive.metastore.client.capability.check` to the underlying Hive client in the Hive catalog if you add the `gravitino.bypass.` prefix to the configuration.

Reviewer: This is ... interesting.

"bypass" means "ignore", "omit" or "neglect".

> Gravitino also supports loading catalog-specific configurations from external files. You just need to put your `hive-site.xml` file in the `catalogs/hive/conf` directory, and Gravitino will load it automatically.

Reviewer: Where are these catalog-specific configurations located?
Is there shared location and/or naming convention?

Author: Good questions. This is my first time to contribute the doc. I am not familiar with the whole doc set. Maybe add it later when I go through the whole doc website. Is it OK?

### Start the Gravitino server

- Start the Gravitino server

    After configuring the Gravitino server, run the following command to start the Gravitino server:

    ```shell
    ./bin/gravitino.sh start

>     ```

Reviewer: Please ALWAYS try avoid indentation by 4 spaces.
In markdown, 4-spaces indentation might be implicitly treated as code snippets.
It is annoying.

Author: if using 2 spaces for indentation under an ordered list or unordered list, it looks the content is not indented properly when previewing in local, as below
<img width="1162" alt="image" src="https://github.com/user-attachments/assets/9d59bd94-a6b7-44db-92eb-152b0e80e45c">

Reviewer: For ordered list, the suggested indentation would be 3 spaces, i.e.

````
1. text goes here
   and continues here

   ```shell
   curl -v ...
   ```

   other texts
1. This is the next item
   with long text paragraph...
````

> - Start the Gravitino server with the Web-based GUI

Reviewer: Web-based GUI -> Web UI ?

    By default, the Gravitino server also provides a Web-based GUI on port `8090`. Run the following command to start the Gravitino server with the Web-based GUI:

    ```shell
    ./bin/gravitino.sh run
    ```

After starting the Gravitino server with the Web-based GUI, visit `http://localhost:8090` in your browser to access the Gravitino server through the Web-based GUI. Or, you can run the following command to verify that the Gravitino server is running:

```shell
curl -v -X GET -H "Accept: application/vnd.gravitino.v1+json" -H "Content-Type: application/json" http://localhost:8090/api/version
```

:::info

> If you need to debug the Gravitino server, enable the `GRAVITINO_DEBUG_OPTS` environment variable in the `conf/gravitino-env.sh` file. Then create a `Remote JVM Debug` configuration in `IntelliJ IDEA` and debug `gravitino.server.main`.

Reviewer: "Enable" an environment variable?

For details about the Gravitino Iceberg REST server, see the [Gravitino Iceberg REST server documentation](./iceberg-rest-service.md).

## Install Apache Gravitino using Docker

Gravitino publishes the Docker image to [Docker Hub](https://hub.docker.com/r/apache/gravitino/tags). The published Gravitino Docker image only contains the Gravitino server with basic configurations.

### Prerequisites

If you have not installed Docker, download and install it by following the [instructions](https://docs.docker.com/get-started/get-docker/) for your Operating System (OS).

### Steps

1. Start the Gravitino server.

    ```shell
    docker run -d -i -p 8090:8090 apache/gravitino:<version>
    ```

2. Visit `http://localhost:8090` in your browser to access the Gravitino server through the Web-based GUI. Or, you can run the following command to verify that THE Gravitino server is running.

    ```shell
    curl -v -X GET -H "Accept: application/vnd.gravitino.v1+json" -H "Content-Type: application/json" http://localhost:8090/api/version
    ```

## Install Apache Gravitino using Docker Compose

> If you want to experience the whole Gravitino system with other components, use the Docker Compose.

Reviewer: ```suggestion
If you want to try out the whole Gravitino system with other components, use the Docker Compose.
```

## Install Apache Gravitino from scratch

### Prerequisites

If you have not installed Java, install it and configure the `JAVA_HOME` environment variable.

> After installation, you can run the `${JAVA_HOME}/bin/java -version` command to check the Java version.

Reviewer 2: Perhaps just `java -version`?

Author: This is the command of the original doc. I did not change the command. If you are sure, we can update it. If not, leave it here to keep it as before.

Reviewer 2: leaving it is fine

:::note

> Gravitino supports running on Java 8, 11, and 17.

Reviewer 2: It would be best to mention the version required before asking them to install it?

Author: whose version? Gravitino's version?

Reviewer 2: The Java version. People are likely to install the latest Java, not Java 8, 11 or 17, if they don't have it.

:::

### Get the Gravitino package

The Gravitino package comprises the Gravitino server and the Gravitino Iceberg REST server.
You can manage these servers independently or run them concurrently on a single server.

> 1. Download the latest Gravitino package from [GitHub](https://github.com/apache/gravitino/releases).

Reviewer 2: We should direct users to the official ASF distribution page first https://gravitino.apache.org/downloads

   :::note
   You can also build Gravitino by following the instructions in [How to Build Gravitino](./how-to-build.md).

   - If you build Gravitino using the `./gradlew compileDistribution` command, you can find the Gravitino package in the `distribution/package` directory.

   - If you build Gravitino using the `./gradlew assembleDistribution` command, you can get the compressed Gravitino package with the name `gravitino-<version>-bin.tar.gz` in the `distribution` directory with sha256 checksum file `gravitino-<version>-bin.tar.gz.sha256`.
   :::

> 2. Extract the Gravitino package and you should have these directories and files:

Reviewer 2: Perhaps this is better? "After extracting the Gravitino package, you should see the following directories and files:"

Author: From my understanding, after.... is a result of an action not a step. here I use the word and to show the sequence of two actions.

Reviewer 2: Fine to keep as is.

      |── conf/                                   # All configurations for Gravitino.
      |   ├── gravitino.conf                      # Gravitino server and Gravitino Iceberg REST server configuration.
      |   ├── gravitino-iceberg-rest-server.conf  # Gravitino server configuration.
      |   ├── gravitino-env.sh                    # Environment variables, etc., JAVA_HOME, GRAVITINO_HOME, and more.
      |   └── log4j2.properties                   # log4j configuration for the Gravitino server and Gravitino Iceberg REST server.
      |── libs/                                   # Gravitino server dependencies libraries.
      |── logs/                                   # Gravitino server and Gravitino Iceberg REST server logs. Automatically created after the server starts.
      |── data/                                   # Default directory for the Gravitino server to store data.
      |── iceberg-rest-server/                    # Gravitino Iceberg REST server package and dependencies libraries.
      └── scripts/                                # Extra scripts for Gravitino.
   ```

### Initialize the RDBMS (Optional)

> If you want to use the `relational` backend storage, you need to initialize the RDBMS.

Reviewer 2: With the removal of the K-V store, I think this is no longer an optional step and would be required but not 100% sure.

Author: let's leave it here for now to see if someone can confirm it.

For details on initializing the RDBMS, see [How to use relational backend storage](./how-to-use-relational-backend-storage.md).

### Configure the Gravitino server

The `conf/gravitino.conf` file provides basic configurations for the Gravitino server.

> To configure the Gravitino server, you can update this file.

Reviewer 2: Perhaps this is better?
"You can modify this file to adjust the Gravitino server settings according to your needs."

For a full list of the Gravitino server configurations, see [Gravitino Server Configurations](./gravitino-server-config.md).

### Configure the Gravitino server log

Gravitino uses [Log4j2](https://logging.apache.org/log4j/2.x/) as the logging system.
The `conf/log4j2.properties` file provides the Gravitino server log configurations.

> To configure the Gravitino server log, you can update this file.

Reviewer 2: I'd use something like:
The logging configuration for the Gravitino server is defined in the `conf/log4j2.properties` file. To customize the logging behavior, modify this file as needed.

### Configure the Gravitino server environment variables

Gravitino exposes several environment variables through the `conf/gravitino-env.sh` file.

> To configure these environment variables, you can update this file.

Reviewer 2: Perhaps something like:
Gravitino provides a set of environment variables defined in the `conf/gravitino-env.sh` file. You can configure these environment variables by editing this file.

### Configure the Gravitino catalogs

Gravitino supports multiple catalogs.
You can configure the catalog-level configurations by updating the related configuration file in the `catalogs/<catalog-provider>/conf` directory.
For example, the `catalogs/hive/conf/hive.conf` file provides configurations for the Hive catalog.
The configurations you set in the catalog configuration file will apply to all the catalogs of the same type that you created.

> For detailed configurations about each catalog, see related catalog documentation.

Reviewer 2: There are some minor issues here with words like configurations and created.

Gravitino supports multiple catalogs, each with its own configuration.
You can configure catalog-level settings by updating the appropriate configuration file in the `catalogs/<catalog-provider>/conf` directory.
For example, the `catalogs/hive/conf/hive.conf` file contains the configuration options for the Hive catalog.
The settings specified in a catalog configuration file will apply to all catalogs of the same type that you create.
 Check the related catalog documentation for detailed information about configuring each catalog,

:::note

> Gravitino takes the catalog configurations in the following order of precedence:

Reviewer 2: Gravitino applies catalog configuration in the following order of precedence:

1. Catalog `properties` specified in the catalog creation API or REST API.
2. Catalog configurations specified in the catalog configuration file.

> The catalog `properties` overrides the catalog configurations specified in the configuration file.

Reviewer 2: 2. Catalog configuration file settings defined in the catalog configuration file. 
Catalog `properties` override the configuration defined in the catalog configuration file.

Author: 1. why the second uses catalog configuration file settings ... instead of catalog configurations

Reviewer 2: The word configurations is misused. The noun configuration can be countable or uncountable, i.e. the plural form of configuration is configuration. You might use configurations to refer to multiple configuration files, but even that is quite rare.

:::

> Gravitino supports passing in catalog-specific configurations by adding the `gravitino.bypass.` prefix to the specific catalog configurations.

Reviewer 2: configurations -> configuration

Author: why use singular from? do you mean  Gravitino supports passing in only one configuration?

Reviewer 2: See above. Configuration is generally the plural of configuration, and it is uncommon to use "configurations".

For example, you can pass in the HMS-specific configuration `hive.metastore.client.capability.check` to the underlying Hive client in the Hive catalog if you add the `gravitino.bypass.` prefix to the configuration.

Gravitino also supports loading catalog-specific configurations from external files.
You just need to put your `hive-site.xml` file in the `catalogs/hive/conf` directory, and Gravitino will load it automatically.

> ### Start the Gravitino server

Reviewer 2: Starting the Gravitino server?

Author: using verb or verb-ing is related to the writing style of the whole doc. The original doc use a verb + noun format. Any special reason that you want to use the verb-ing format?

Reviewer 2: No hard preference either way, just a suggestion to consider.

- Start the Gravitino server

>   After configuring the Gravitino server, run the following command to start the Gravitino server:

Reviewer 2: Once you've configured the Gravitino server, use the following command to start it:

   ```shell
   ./bin/gravitino.sh start
   ```

- Start the Gravitino server with the Web UI

  By default, the Gravitino server also provides a Web UI on port `8090`.
  Run the following command to start the Gravitino server with the Web UI:

   ```shell
   ./bin/gravitino.sh run
   ```

After starting the Gravitino server with the Web UI, visit `http://localhost:8090` in your browser to access the Gravitino server through the Web UI.

> Or, you can run the following command to verify that the Gravitino server is running:

Reviewer 2: verify -> check?

Author: I think both words are ok.

Reviewer 2: Check is less formal, and verify implies more work. While both are OK, it seems a little odd to me as a native English speaker to use verify in this context.

```shell
curl -v -X GET -H "Accept: application/vnd.gravitino.v1+json" -H "Content-Type: application/json" http://localhost:8090/api/version
```

:::info
If you need to debug the Gravitino server, configure the `GRAVITINO_DEBUG_OPTS` environment variable in the `conf/gravitino-env.sh` file.
Then create a `Remote JVM Debug` configuration in `IntelliJ IDEA` and debug `gravitino.server.main`.
:::

> ### Start the Gravitino Iceberg REST server

Reviewer 2: Starting?

> You can run the Iceberg REST server either as a standalone server or as an auxiliary service embedded in the Gravitino server.

Reviewer 2: remove auxiliary

- To start the Iceberg REST server as a standalone server, run the `./bin/gravitino-iceberg-rest-server.sh start` command with configurations specified in the `./conf/gravitino-iceberg-rest-server.conf` file.

> - To start the Iceberg REST server as an auxiliary service embedded in the Gravitino server, run the `./bin/gravitino.sh start` command with all configurations in the `conf/gravitino.conf` file.

Reviewer 2: remove auxiliary and change configurations. for example:
To start the Iceberg REST server as an embedded service within the Gravitino server, run the `./bin/gravitino.sh` start command. The configuration in the `conf/gravitino.conf` file will be applied.

"will be applied" is perhaps a bit too formal "will be used" might be better?

For details about the Gravitino Iceberg REST server, see the [Gravitino Iceberg REST server documentation](./iceberg-rest-service.md).

> ## Install Apache Gravitino using Docker

Reviewer 2: Installing?

Gravitino publishes the Docker image to [Docker Hub](https://hub.docker.com/r/apache/gravitino/tags).

> The published Gravitino Docker image only contains the Gravitino server with basic configurations.

Reviewer 2: with basic configurations -> with a basic configuration

Author: do you mean that there is only one config item in the docker image?

Reviewer 2: Multiple config items are still called a configuration.

### Prerequisites

> If you have not installed Docker, download and install it by following the [instructions](https://docs.docker.com/get-started/get-docker/) for your Operating System (OS).

Reviewer 2: no need for "(OS)"

Author: OS is an abbreviation of operating system. This is common used for technical writing. any special reason why do you want to use operating system?

Reviewer 2: It is unnecessary here as it is not used anywhere else. Usually, you would introduce an abbreviation without capitalisation e.g. "operating system (OS)" and then use OS in the rest of the documents.

### Steps

1. Start the Gravitino server.

   ```shell
   docker run -d -i -p 8090:8090 apache/gravitino:<version>
   ```

2. Visit `http://localhost:8090` in your browser to access the Gravitino server through the Web UI.

> Or, you can run the following command to verify that THE Gravitino server is running.

Reviewer 2: THE -> the

   ```shell
   curl -v -X GET -H "Accept: application/vnd.gravitino.v1+json" -H "Content-Type: application/json" http://localhost:8090/api/version
   ```

> ## Install Apache Gravitino using Docker Compose

Reviewer 2: Installing the?

> If you want to try out the whole Gravitino system with other components, use the Docker Compose.

Reviewer 2: use the Docker Compose - > use Docker Compose.
